### PR TITLE
Rework mobile server release notification and mismatch message

### PR DESCRIPTION
**Pull request by ddshd**
- Only shows release notification for logged in admin users
- Updated version outdated message in the menu bar with translation support

This addresses either a discussion post or discord thread (I'll try to find it again) where they didn't want their family members / user to be worried about the outdated server due to the release announcement.

# How was this tested:
An old server version was used.
### Simulator memory cleared before every test:
- Launch app (No release announcement shown)
- Login using non-admin user (No release announcement shown, server outdated menu message still shown)
- Login using admin user (Release announcement shown, server outdated menu message shown)

### WIthout clearing simulator memory
- Log out of non-admin user and log into admin user (Release announcement shown, server outdated menu message shown)
- Log out of admin user and log into non-admin user (No release announcement shown, server outdated menu message still shown)

### CODE REVIEW

File: mobile/lib/shared/providers/release_info.provider.dart [hunk 1]
```diff
@@ -23,35 +41,70 @@ class ReleaseInfoNotifier extends StateNotifier<String> {
...
+          patch_: 0,
+          minor: 0,
+        ),
+        isVersionMismatch: false,
+        versionMismatchErrorMessage: "",
+      );
     }
   }
 
+  String serverVersionReponseToString(ServerVersionResponseDto dto) {
```
Comment (alextran1502) [inline mobile/lib/shared/providers/release_info.provider.dart:97]: Can you refactor this method to a common util file to now have duplicated code in two places? also there is typo in the function name

Comment (ddshd) [reply mobile/lib/shared/providers/release_info.provider.dart]: Sorry about that.. Done!

**Comment by shenlong-tanwen** [comment]
@ddshd Is it possible for you to rebase this over the current main and update the handling to remove the update alert all together? We have a warning badge displayed near the user profile icon in the app bar and also a mismatch message in the server info panel. It might not be necessary to also show an update message to the users.

**Comment by jrasm91** [comment]
Closing, as it has been implemented in #5104.

**Pull request closed without merging**